src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Parameters/SearchParameterValidator.cs

```diff
@@ -141,13 +142,47 @@ private void CheckForConflictingCodeValue(SearchParameter searchParam, List<Vali
                             nameof(searchParam.Code),
                             string.Format(Resources.SearchParameterDefinitionNullorEmptyCodeValue, searchParam.Code, baseType.ToString())));
                 }
-                else if (_searchParameterDefinitionManager.TryGetSearchParameter(baseType.ToString(), searchParam.Code, out _))
+                else
                 {
-                    // The search parameter's code value conflicts with an existing one
-                    validationFailures.Add(
+                    if (baseType.ToString() == KnownResourceTypes.Resource)
```

Comment:
Specify StringComparison for consistency, I'd use Ordinal in this case for byte level comparison:
```suggestion
                    if (string.Equals(baseType.ToString(), KnownResourceTypes.Resource, StringComparison.OrdinalIgnoreCase))
```